File: sort/insertion_sort/haskell/InsertionSortTest.hs
module InsertionSortTest.Main (main) where
import Data.List (sort)

import qualified InsertionSort as StdLibIns
import qualified InsertionSortWithCustomInsert as CustomIns

main :: IO ()
main = do
    putStr "Standard library `insert`: "
    test StdLibIns.insertionSort
    putStr "Custom `insert`: "
    test CustomIns.insertionSort

test :: ([Int] -> [Int]) -> IO ()
test f = if f list == sort list
    then putStrLn "List sorted correctly"
    else putStrLn "Error: List did not sort correctly"
  where list = [12,3,55,1,2,7,4]

File: sort/insertion_sort/java/InsertionSortTest.java
import org.junit.Test;
import java.util.Arrays;

public class InsertionSortTest {

    @Test
    public void test() {
        int list[] = { -5, 8, 0, 12, 55, 1, -66, 35, 3 };
        InsertionSort insertionSort = new InsertionSort();
        insertionSort.sort(list);
        for (int i = 0; i < list.length - 1; i++) {
            assert (list[i] <= list[i + 1]);
        }
    }

    @Test
    public void nullList(){
        int list[] = {};
        int expectedList[] = {};
        InsertionSort insertionSort = new InsertionSort();
        insertionSort.sort(list);
        assert (Arrays.equals(list, expectedList));
    }
}
